File: debezium/connector/embedded/EmbeddedRunner.java
package com.tiket.tix.hotel.common.debezium.connector.embedded;

import com.tiket.tix.hotel.common.debezium.DebeziumHelper;
import com.tiket.tix.hotel.common.debezium.connector.ConnectorRunner;

public class EmbeddedRunner implements ConnectorRunner {

  private final EmbeddedConnectorRunner runner;

  public EmbeddedRunner(EmbeddedConnectorRunner runner) {
    this.runner = runner;
  }

  @Override
  public void start() {
    runner.startEmbedded();
    DebeziumHelper.sendMetric("connector_started_use_embedded");
  }
}

File: debezium/connector/embedded/embedded_runner.go
package embedded

import (
	"github.com/tiket/TIX-HOTEL-UTILITIES-GO/debezium/helper"
)

type EmbeddedRunner struct {
	runner *UpsertConnectorRunner
}

func NewEmbeddedRunner(runner *UpsertConnectorRunner) *EmbeddedRunner {
	return &EmbeddedRunner{runner: runner}
}

func (e *EmbeddedRunner) Start() error {
	e.runner.StartEmbedded()
	// metric placeholder
	helper.SendMetric(helper.TagBuilder{
		EventType: "connector_started_use_embedded",
	})
	return nil
}
